### ISSUE TITLE
CSP: Safari 8.x, 9, and 9.1 does not support source expressions with full paths

### ISSUE THREAD

**Issue by papandreou** [assetgraph/assetgraph#635]
CSP level 1 (https://www.w3.org/TR/2012/CR-CSP-20121115/#source-list) foresaw the need to support whitelisting of external resources more specifically than by origin with an optional protocol, so they explicitly defined `ext-host-source` like this:

```
host-source       = [ scheme "://" ] host [ port ]
ext-host-source   = host-source "/" *( <VCHAR except ";" and ","> )
                  ; ext-host-source is reserved for future use.
```

... and in the next section it says that it should be treated the same way as `host-source`:

```
If the source expression matches the grammar for host-source or ext-host-source:
```

... so CSP1 compliant implementations should be compatible with the changes in https://github.com/assetgraph/assetgraph/pull/618 by ignoring the path, thus treating it as a whitelisting of the CDN hostname.

However, Safari 8 and 9 simply ignore source expressions matching the `ext-host-source` production, so we don't have a proper fallback strategy for that case. That means that Safari 8 and 9 won't realize that the CDN is to be whitelisted for `script-src` and `style-src`.

In the app where I'm currently implementing CSP I've fixed that by whitelisting the CDN hostname in `script-src` and `style-src`, thus negating the security gains from https://github.com/assetgraph/assetgraph/pull/618

I think we're left with two possibilities:

1) Revert https://github.com/assetgraph/assetgraph/pull/618
2) Only apply https://github.com/assetgraph/assetgraph/pull/618 when Safari 8, 9, and 9.1 aren't in the set specified by `--browsers`.

**Comment by papandreou** [comment]
Convinced myself to implement solution 2). Fixed in 3.0.0-30/ag-b v5.0.0-23.

**Comment by Munter** [comment]
Yeah, that sounded like the better one to me too
